modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -156,10 +163,56 @@ function ding_adgangsplatformen_openplatform_token_get() {
   return $token;
 }
 
+/**
+ * Provider function.
+ *
+ * Get token for authentication as the library. Should also be used when the
+ * service called don't need access to user information or if no user is logged
+ * in.
+ */
+function ding_adgangsplatformen_openplatform_get_agency_token() {
+  $token = &drupal_static(__FUNCTION__);
+  if (!isset($token)) {
+    $cache = cache_get('ddb_cover_upload_token');
+    if ($cache && $cache->expire > REQUEST_TIME) {
+      $token = $cache->data;
+    }
+    else {
+      $config = ding_adgangsplatformen_get_configuration();
+      $agency_id = variable_get('ting_agency', '');
+
+      $client = new \GuzzleHttp\Client();
+      $response = $client->request('POST', $config['urlAccessToken'], [
+        'form_params' => [
+          'grant_type' => 'password',
+          'username' => '@' . $agency_id,
+          'password' => '@' . $agency_id,
+        ],
+        'auth' => [
+          $config['clientId'],
+          $config['clientSecret'],
+        ],
+      ]);
+
+      $content = $response->getBody()->getContents();
+      $json = json_decode($content, true);
+
+      $token = $json['access_token'];
```

Comment:
Have you considered validating the response? What if OpenPlatform returns unexpected values?

Reply:
Checks added.